dao/GlossaryDAO.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Glossary\DAO;
+
+class GlossaryDAO {
+        
+    private $PDOX;
+    private $p;
+
+    public function __construct($PDOX, $p) {
+        $this->PDOX = $PDOX;
+    }
+   
+    // getAllDomains
+    // returns glossary_domains
+    function getDomain($domain_id = 0, $limit = 0, $offset = 0) {   
+        $arr = array();
+        $query = "SELECT *
+                    FROM `{$this->p}glossary_domain` `domain`";
+                    
+        if ($domain_id > 0) {
+            $query .= "WHERE `domain`.`id`=:domain_id";
+            $arr = array(':domain_id' => $domain_id);
+        }
+
+        return $this->PDOX->allRowsDie($query, $arr);
+    }
+    function getAlphabetTerms($link_id, $alphabet) {    
+         $query = "SELECT `id`, `domain_id`, `term`, `description`  
+                    FROM `{$this->p}glossary_term` 
+                    WHERE  `term` LIKE :alphabetPrefix";     
+                    $arr = array(   ':alphabetPrefix' => $alphabet . '%'    );     return $this->PDOX->allRowsDie($query, $arr); }
+    function getTermsForDomain($domain_id = 0, $limit = 0, $offset = 0) {   
+        $arr = array();
+        $query = "SELECT `def`.`id`, `domain`.`name` as `domain_name`,`def`.`term`,`def`.`description` 
+                    FROM `{$this->p}glossary_term` `def` 
+                    LEFT JOIN `{$this->p}glossary_domain` `domain` on `domain`.`id` = `def`.`domain_id`";
+
+        if ($domain_id > 0) {
```

Comment:
Deleted = 0 is used in both cases so this can be simplified from:
```
        $query = "SELECT `def`.`id`, `domain`.`name` as `domain_name`,`def`.`term`,`def`.`description` 
            FROM `{$this->p}glossary_term` `def` 
            LEFT JOIN `{$this->p}glossary_domain` `domain` on `domain`.`id` = `def`.`domain_id`";
        if ($domain_id > 0) {
            $query .= "WHERE `domain_id`=:domain_id AND `deleted` = 0";
            $arr = array(':domain_id' => $domain_id);
        } else {
            $query .= "WHERE `deleted` = 0";
        }
```
To:
```
        $query = "SELECT `def`.`id`, `domain`.`name` as `domain_name`,`def`.`term`,`def`.`description` 
            FROM `{$this->p}glossary_term` `def` 
            LEFT JOIN `{$this->p}glossary_domain` `domain` on `domain`.`id` = `def`.`domain_id`
            WHERE `deleted` = 0 ";
        if ($domain_id > 0) {
            $query .= "and `domain_id`=:domain_id";
            $arr = array(':domain_id' => $domain_id);
        }
```